packages/core/src/components/toast/overlayToaster.tsx

```diff
@@ -152,24 +167,66 @@ export class OverlayToaster extends AbstractPureComponent<OverlayToasterProps, O
             return options.key;
         }
 
+        if (this.queue.isRunning) {
+            // If a toast has been shown recently, push to the queued toasts to prevent toasts from being shown too
+            // quickly for the animations to keep up
+            this.queue.toasts.push(options);
+        } else {
+            // If we have not recently shown a toast, we can immediately show the given toast
+            this.immediatelyShowToast(options);
+            this.maybeStartQueueTimeout();
+        }
+
+        return options.key;
+    }
+
+    private maybeUpdateExistingToast(options: ToastOptions, key: string | undefined) {
+        if (key == null) {
+            return false;
+        }
+
+        const isExistingQueuedToast = this.queue.toasts.some(toast => toast.key === key);
+        if (isExistingQueuedToast) {
+            this.queue.toasts = this.queue.toasts.map(t => (t.key === key ? options : t));
+            return true;
+        }
+
+        const isExistingShownToast = this.state.toasts.some(toast => toast.key === key);
+        if (isExistingShownToast) {
+            this.updateToastsInState(toasts => toasts.map(t => (t.key === key ? options : t)));
+            return true;
+        }
+
+        return false;
+    }
+
+    private immediatelyShowToast(options: ToastOptions) {
         if (this.props.maxToasts) {
             // check if active number of toasts are at the maxToasts limit
             this.dismissIfAtLimit();
         }
 
         this.updateToastsInState(toasts => [options, ...toasts]);
-        return options.key;
     }
 
-    private maybeUpdateExistingToast(options: ToastOptions, key: string | undefined) {
-        if (key == null || this.isNewToastKey(key)) {
-            return false;
+    private maybeStartQueueTimeout() {
+        if (this.queue.isRunning) {
+            return;
         }
 
-        this.updateToastsInState(toasts => toasts.map(t => (t.key === key ? options : t)));
-        return true;
+        this.queue.isRunning = true;
+        this.queue.cancel = this.setTimeout(this.handleQueueTimeout, QUEUE_TIMEOUT_MS);
     }
 
+    private handleQueueTimeout = () => {
+        const nextToast = this.queue.toasts.shift();
+        this.queue.isRunning = false;
```

Comment:
Do we have to set `isRunning` to `false` during each execution of the timeout? Would it make sense to check that the queue is empty first? Something like:
```ts
if (this.queue.toasts.length === 0) {
    this.queue.isRunning = false;
}
```

Reply:
Yeah good call. I ended up implementing this slightly differently but with the same intent.